rust_src/src/textprop.rs

```diff
@@ -28,7 +29,39 @@ pub fn get_char_property(position: EmacsInt, prop: LispObject, object: LispObjec
 /// If POSITION is at the end of OBJECT, the value is nil.
 #[lisp_fn(min = "2")]
 pub fn get_text_property(position: EmacsInt, prop: LispObject, object: LispObject) -> LispObject {
-    unsafe { textget(Ftext_properties_at(position.into(), object), prop) }
+    unsafe { textget(text_properties_at(position.into(), object), prop) }
+}
+
+/// Return the list of properties of the character at POSITION in OBJECT.
+/// If the optional second argument OBJECT is a buffer (or nil, which means
+/// the current buffer), POSITION is a buffer position (integer or marker).
+/// If OBJECT is a string, POSITION is a 0-based index into it.
+/// If POSITION is at the end of OBJECT, the value is nil.
+#[lisp_fn(min = "1")]
+pub fn text_properties_at(mut position: LispObject, mut object: LispObject) -> LispObject {
```

Comment:
I think you can make position a `LispNumber`.

Reply:
I managed it but ended up breaking a lot more tests than before :). I'm sure it will be possible once validate_interval_range is ported.